File: Libraries/python/beagle_helper/src/beagle_helper/devices/pwm_led.py
from ..pin import Pin

class PwmLed:
    def __init__(self, pin: Pin) -> None:
        self.pwm = pin.pwm_output()

    def start(self, duty_cycle: int, period: int = 255):
        self.pwm.set_period(period)
        self.pwm.set_duty_cycle(duty_cycle)
        self.pwm.enable()

File: Libraries/rust/beagle_helper/src/devices/pwm_led.rs
use crate::abstractions::pwm::PwmChannel;

pub struct PwmLed(PwmChannel);

impl PwmLed {
    pub fn new(pin: crate::pin::Pin) -> std::io::Result<Self> {
        pin.pwm_output().map(Self)
    }

    pub fn start(&self, duty_cycle: usize, period: usize) -> std::io::Result<()> {
        self.0.set_period(period)?;
        self.0.set_duty_cycle(duty_cycle)?;
        self.0.enable()
    }
}
